src/Grids/rectilinear_grid.jl

```diff
@@ -253,7 +253,7 @@ RectilinearGrid{Float64, Periodic, Bounded, Bounded}
                 grid in z: Stretched, with spacing min=0.6826950100338962, max=1.8309085743885056
 ```
 """
-function RectilinearGrid(architecture = CPU(),
+function RectilinearGrid(architecture::AbstractArchitecture = CPU(),
                          FT = Float64;
```

Comment:
```suggestion
                         FT::DataType = Float64;
```

Reply:
It's not needed for dispatch, but could help prevent bugs (though this usage violates YAS...)

Though typing on `architecture` also violates YAS (it's also not needed for dispatch).

Reply:
@navidcy ok to merge without this?

Reply:
> @navidcy ok to merge without this?

Sure!!